src/Controller/Admin/FacetsController.php

```diff
@@ -37,9 +37,16 @@ public function fieldRowAction()
     public function fieldEditSidebarAction()
     {
         $fieldData = $this->params()->fromQuery('field_data');
+        $searchPageId = $this->params()->fromQuery('search_page_id');
+        $searchPage = $this->api()->read('search_pages', $searchPageId)->getContent();
+
         $form = $this->getForm(FacetForm::class);
         $form->setData($fieldData);
 
+        $index = $searchPage->index();
+        $facetSorts = $index->availableFacetSorts();
+        $form->setFacetSorts($facetSorts);
```

Comment:
I think it would be best to pass the `$searchPage` as an option to the FacetForm (using the 2nd parameter of `getForm`). That way the form can be fully initiated in the init() function, and the form logic remains in the form class.